prediction-insights.ipynb

```diff
@@ -0,0 +1,774 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "The [basic workflow guide](https://docs.howso.com/user_guide/basics/basic_workflow.html) goes into more specifics about the individual details of this section. This recipe will focus more on the insights.\n",
+    "\n",
+    "### Step 1: Load Data and Create Trainee"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "The following parameters from configuration file will override the Amalgam parameters set in the code: {'trace'}\n"
+     ]
+    }
+   ],
+   "source": [
+    "df = fetch_data(\"iris\", local_cache_dir=\"data\")\n",
+    "features = infer_feature_attributes(df)\n",
+    "\n",
+    "t = Trainee(features=features)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Step 2: Train, Analyze, and React"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "t.train(df)\n",
+    "t.analyze()\n",
+    "t.react_into_features(similarity_conviction=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Step 3: React to a Case\n",
+    "\n",
+    "We supply a case using the `context_features` and `context_values` parameters. These are treated as the independent\n",
+    "variables when making the prediction. The `action_features` describe the dependent variables and are what are output\n",
+    "in addition to any requested details."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "reaction = t.react(\n",
+    "    contexts=[3.0, 2.0, 3.0, 1.0],\n",
+    "    context_features=[\"sepal-length\", \"sepal-width\", \"petal-length\", \"petal-width\"],\n",
+    "    action_features=[\"target\"],\n",
+    "    details={\n",
+    "        \"influential_cases\": True,\n",
+    "        \"similarity_conviction\": True,\n",
+    "        \"feature_contributions\": True,\n",
+    "        \"feature_residuals\": True,\n",
+    "        \"robust_influences\": True,\n",
+    "        \"robust_residuals\": True,\n",
+    "        \"local_case_feature_residual_convictions\": True,\n",
+    "        \"categorical_action_probabilities\": True,\n",
+    "    }\n",
+    ")\n",
+    "influential_cases = pd.DataFrame(reaction[\"details\"][\"influential_cases\"][0])\n",
+    "influential_cases = influential_cases[sorted(influential_cases.columns)]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Insights\n",
+    "\n",
+    "### Insight 1: Prediction\n",
+    "\n",
+    "The predicted value(s) are accessed using the `action` key, while the explanations can be accessed using the\n",
```

Comment:
I don't think this is necessary. Let's just show the prediction without the details as to how to call it.

For example we can say: "here is your prediction, we can see that from our inputs, Howso Engine has inferred that the iris type is : xyz"